sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -27,6 +27,10 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     uint8_t type = (flags & IEEE802154_FCF_TYPE_MASK);
     uint8_t bcast = (flags & IEEE802154_BCAST);
 
+    if (dst_pan.u16 == src_pan.u16) {
```

Comment:
to be on the safe side: could you also clear the bit in case the pans are different?